Doc/howto/clinic.rst

```diff
@@ -93,67 +96,143 @@ and it should be able to do many interesting and smart
-generates output.  This output is rewritten into the C file
-immediately after the block, followed by a comment containing a checksum.
-The Argument Clinic block now looks like this:
+Everything in between these two lines is Argument Clinic :term:`input`.
+When Argument Clinic parses input, it generates :term:`output`.
+The output is rewritten into the C file immediately after the input,
+followed by a :term:`checksum line`.
+All of these lines, including the :term:`start line` and :term:`checksum line`,
+are collectively called an Argument Clinic :term:`block`:
 
-.. code-block:: none
+.. code-block:: c
 
     /*[clinic input]
     ... clinic input goes here ...
     [clinic start generated code]*/
     ... clinic output goes here ...
-    /*[clinic end generated code: checksum=...]*/
+    /*[clinic end generated code: ...]*/
 
 If you run Argument Clinic on the same file a second time, Argument Clinic
-will discard the old output and write out the new output with a fresh checksum
-line.  However, if the input hasn't changed, the output won't change either.
-
-You should never modify the output portion of an Argument Clinic block.  Instead,
-change the input until it produces the output you want.  (That's the purpose of the
-checksum—to detect if someone changed the output, as these edits would be lost
-the next time Argument Clinic writes out fresh output.)
-
-For the sake of clarity, here's the terminology we'll use with Argument Clinic:
-
-* The first line of the comment (``/*[clinic input]``) is the *start line*.
-* The last line of the initial comment (``[clinic start generated code]*/``) is the *end line*.
-* The last line (``/*[clinic end generated code: checksum=...]*/``) is the *checksum line*.
-* In between the start line and the end line is the *input*.
-* In between the end line and the checksum line is the *output*.
-* All the text collectively, from the start line to the checksum line inclusively,
-  is the *block*.  (A block that hasn't been successfully processed by Argument
-  Clinic yet doesn't have output or a checksum line, but it's still considered
-  a block.)
+will discard the old :term:`output` and write out the new output with a fresh
+:term:`checksum line`.
+If the :term:`input` hasn't changed, the output won't change either.
+
+.. note::
+
+   You should never modify the output of an Argument Clinic block,
+   as any change will be lost in future Argument Clinic runs;
+   Argument Clinic willl detect an output checksum mismatch and regenerate the
+   correct output.
+   If you are not happy with the generated output,
+   change instead the input until it produces the output you want.
+
+
+.. _clinic-reference:
+
+Reference
+=========
+
+
+.. _clinic-terminology:
+
+Terminology
+-----------
+
+.. glossary::
+
+  start line
+     The line ``/*[clinic input]``.
+     This line marks the beginning of Argument Clinic input.
+
+  end line
+     The line ``[clinic start generated code]*/``.
```

Comment:
Is this an error or just confusingly named?

Reply:
It's confusingly named; the opening line is the start of a C comment; the end line is the end of that comment. Argument Clinic input lives inside a C comment. Perhaps we should mention that.

Reply:
> Perhaps we should mention that.

How about 951c92cf1372f95382fc27fcd6f8605c75f93b78?

Reply:
oof, I still find this quite confusing. Could we state explicitly in this definition why it says "start generated code", even though it's called the "end line"? (It's the _end_ of the _input_, but also the _start_ of the _output_).

Reply:
Or maybe we could rename this term "input end line"...?

Reply:
Thanks for pointing this out! I'm so used to clinic code, so for _me_ it's not confusing at all that the **end line** is `clinic start generated code*/` 😅 I'll think of something and revisit this.

Reply:
See 72fe212adb9b87609ac5ca1232472e87acb749bf